.github/workflows/build-image.yaml

```diff
@@ -0,0 +1,51 @@
+name: Build and Publish
+
+on:
+  push:
+    branches: [ main ]
+    tags: [ 'v*' ]
+    paths:
+      - "**.go"
+      - go.mod
+      - go.sum
+  pull_request:
+    branches: [ main ]
+    paths:
+      - "**.go"
+      - go.mod
+      - go.sum
+
+env:
+  KO_DOCKER_REPO: ghcr.io/${{ github.repository }}/controller
+
+jobs:
+  build-and-push:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      packages: write
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Set up Go
+        uses: actions/setup-go@v4
+        with:
+          go-version: '1.24'
```

Comment:
lets pin to the go.mod instead of hardcoding?

Reply:
TIL about `go-version-file`